### PR TITLE
DDF-2985 Cleanup unused catalog-core-api dependencies

### PR DESCRIPTION
**Pull request by peterhuffer**
#### What does this PR do?
- Removes unused dependencies from `catalog-core-api` pom.
- Updates other poms to explicitly declare their dependencies they were receiving transitively from `catalog-core-api`
- This will break downstream projects if they do not explicitly depend on a dependency they were receiving transitively.

#### Who is reviewing it? 
@emmberk @tbatie @stustison @clockard 

#### Select relevant component teams: 
@codice/build 

#### Ask 2 committers to review/merge the PR and tag them here.
@coyotesqrl
@vinamartin

#### How should this be tested?
Full build.

#### Any background context you want to provide?
Noticed when doing a jackson upgrade that a lot of modules had transitive dependencies on jackson-core library because `catalog-core-api` defined the `org.codice.thirdparty/tika-bundle` dependency.

#### What are the relevant tickets?
[DDF-2985](https://codice.atlassian.net/browse/DDF-2985)
Going to use this ticket and keep it open as a ticket for general cleanup of poms.

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

File: catalog/core/catalog-core-api/pom.xml [hunk 1]
```diff
@@ -30,76 +30,6 @@
...
-                </exclusion>
-                <exclusion>
-                    <groupId>net.minidev</groupId>
-                    <artifactId>asm</artifactId>
-                </exclusion>
-            </exclusions>
-        </dependency>
-        <dependency>
-            <groupId>org.codice.thirdparty</groupId>
-            <artifactId>tika-bundle</artifactId>
```
Comment (emmberk) [inline catalog/core/catalog-core-api/pom.xml]: ❓Does the catalog-core-api feature still need to start the tika-bundle bundle?
https://github.com/codice/ddf/blob/291eebf9d7996a3847f674a6602ef8ab189e1a60/catalog/catalog-app/src/main/resources/features.xml#L31-L41

Comment (peterhuffer) [reply catalog/core/catalog-core-api/pom.xml]: Probably not. I didn't check these, but I imagine some features will fail to startup once I remove that bundle. I'll go through these right now.

Comment (emmberk) [reply catalog/core/catalog-core-api/pom.xml]: I'm not sure if it's worth the effort. I think we should still move forward with the pom changes regardless of whether the feature is modified too:+1:

Comment (peterhuffer) [reply catalog/core/catalog-core-api/pom.xml]: Agreed. There are many incorrect bundles in this feature that many other features are relying on.

**Comment by peterhuffer** [comment]
build now

**Comment by peterhuffer** [comment]
build now

**Comment by peterhuffer** [comment]
tests have been failing intermittently on other PRs
build now

**Comment by LinkMJB** [comment]
@peterhuffer are the tests that are failing consistently the same tests? 🤔 Should a ticket be opened?

**Pull request merged**